src/rpm_ostree/cli_deploy.rs

```diff
@@ -98,25 +94,32 @@ fn invoke_cli_register() -> Result<()> {
 }
 
 /// CLI executor for deploying upgrades.
-fn invoke_cli_deploy(
-    release: Release,
-    allow_downgrade: bool,
-    custom_origin: Option<CustomOrigin>,
-) -> Result<Release> {
+fn invoke_cli_deploy(release: Release, staging_options: StagingOptions) -> Result<Release> {
     fail_point!("deploy_locked_err", |_| bail!("deploy_locked_err"));
     fail_point!("deploy_locked_ok", |_| Ok(release.clone()));
 
     let mut cmd = std::process::Command::new("rpm-ostree");
     match &release.payload {
-        Payload::Pullspec(image) => {
-            cmd.arg("rebase").arg(image).arg("--lock-finalization");
-            if let Some(origin) = custom_origin {
-                cmd.arg("--custom-origin-url")
-                    .arg(origin.url.clone())
-                    .arg("--custom-origin-description")
-                    .arg(origin.description.clone());
+        Payload::Pullspec(reference) => {
+            if let Some(rebase_target) = staging_options.rebase {
+                cmd.arg("rebase").arg("--lock-finalization");
+                match reference.digest() {
+                    Some(digest) => {
+                        cmd.arg(rebase_target.to_string()).arg(digest);
+                    }
+                    // zincati payload are digested oci pullspecs
+                    _ => unreachable!(),
```

Comment:
I think of `unreachable!()` as more something that can't happen given the internal logic of the code itself. This one depends more on input external to us (the graph). Though in practice possibly we would fail earlier somewhere else if the payload wasn't digested...

But anyway, I guess it doesn't hurt to just do e.g. `let digest = reference.digest().ok_or_else(|| anyhow!())?;`.

Reply:
Yeah if the cincinnati data is incorrect it should not go that far, but let's throw an error rather than panicking.